nvflare/app_common/executors/learner_executor.py

```diff
@@ -105,7 +107,11 @@ def train(self, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -
     def submit_model(self, shareable: Shareable, fl_ctx: FLContext) -> Shareable:
         model_name = shareable.get_header(AppConstants.SUBMIT_MODEL_NAME)
         submit_model_result = self.learner.get_model_for_validation(model_name, fl_ctx)
-        if submit_model_result and isinstance(submit_model_result, Shareable):
+        if (
+            submit_model_result
+            and isinstance(submit_model_result, Shareable)
+            and submit_model_result.get_return_code() == ReturnCode.OK
```

Comment:
Again, if the Learner returns a valid Shareable, we should honor it and not changing its return code. We could return (and skip further processing) if Learner's return code is not OK.